boot/bootutil/src/loader.c

```diff
@@ -578,13 +579,20 @@ boot_version_cmp(const struct image_version *ver1,
  *          does not match the slot address.
  */
 static bool
-boot_rom_address_check(struct boot_loader_state *state, size_t slot)
+boot_rom_address_check(struct boot_loader_state *state,
+                       struct slot_usage_t slot_usage[])
 {
-    const struct image_header *hdr = boot_img_hdr(state, slot);
-    uint32_t f_off = boot_img_slot_off(state, slot);
+    uint32_t active_slot;
+    const struct image_header *hdr;
+    uint32_t f_off;
+
+    active_slot = slot_usage[BOOT_CURR_IMG(state)].active_slot;
+    hdr = boot_img_hdr(state, active_slot);
+    f_off = boot_img_slot_off(state, active_slot);
+
     if (hdr->ih_flags & IMAGE_F_ROM_FIXED && hdr->ih_load_addr != f_off) {
         BOOT_LOG_WRN("Image in %s slot at 0x%x has been built for offset 0x%x"\
-                     ", skipping", slot == 0 ? "primary" : "secondary", f_off,
+                     ", skipping", active_slot == 0 ? "primary" : "secondary", f_off,
```

Comment:
Exceeds 80 char.

Reply:
Have we ever discussed our coding style? I don't see a lot of > 80-character lines, but is that actually a requirement we want?

Reply:
You are right I have not seen this rule written down in this project. However the majority of the code is compliant, that is why I thought this rule is applicable. I would vote to not have this requirement.

Reply:
To be consistent with the current line length in the file I change to not exceed 80 chars.